### PR TITLE
removed tenancy from service categories

### PR DESCRIPTION
**Pull request by mattjackson220**
## What does this PR (Pull Request) do?

- [x] This PR is not related to any Issue 

This PR removes tenancy from Service Categories

## Which Traffic Control components are affected by this PR?

- Documentation
- Traffic Ops
- Traffic Portal

## What is the best way to verify this PR?

verify that unit and API tests pass
verify that GET, POST, PUT, and DELETE still work for /service_categories
verify that all tenants can see all service categories
verify that when a service category that has already been assigned to a delivery service is updated that that delivery service is also updated
verify that a service category assigned to a delivery service cannot be deleted

## The following criteria are ALL met by this PR

- [x] This PR includes tests OR I have explained why tests are unnecessary
- [x] This PR includes documentation OR I have explained why documentation is unnecessary
- [x] This PR includes an update to CHANGELOG.md OR such an update is not necessary
- [x] This PR includes any and all required license headers
- [x] This PR ensures that database migration sequence is correct OR this PR does not include a database migration
- [x] This PR **DOES NOT FIX A SERIOUS SECURITY VULNERABILITY** (see [the Apache Software Foundation's security guidelines](https://www.apache.org/security/) for details)

### CODE REVIEW

File: traffic_ops/traffic_ops_golang/routing/routes.go [hunk 1]
```diff
@@ -420,7 +420,7 @@ func Routes(d ServerData) ([]Route, []RawRoute, http.Handler, error) {
 
 		//ServiceCategories
 		{api.Version{3, 0}, http.MethodGet, `service_categories/?$`, api.ReadHandler(&servicecategory.TOServiceCategory{}), auth.PrivLevelReadOnly, Authenticated, nil, 1085181543, noPerlBypass},
-		{api.Version{3, 0}, http.MethodPut, `service_categories/{name}/?$`, api.UpdateHandler(&servicecategory.TOServiceCategory{}), auth.PrivLevelOperations, Authenticated, nil, 306369141, noPerlBypass},
+		{api.Version{3, 0}, http.MethodPut, `service_categories/{name}/?$`, servicecategory.Update, auth.PrivLevelOperations, Authenticated, nil, 306369141, noPerlBypass},
```
Comment (rawlinp) [inline traffic_ops/traffic_ops_golang/routing/routes.go:423]: Why was this change necessary? Looking at `servicecategory.Update`, it doesn't appear to validate or create a changelog entry, both of which this should be doing via `api.UpdateHandler` already.

**Review by rawlinp — CHANGES_REQUESTED** [review]

File: traffic_ops/traffic_ops_golang/routing/routes.go (hunk 1 above)
Comment (ocket8888) [reply traffic_ops/traffic_ops_golang/routing/routes.go:423]: I also just finished testing, and trying to update a Service Category with a name of an SC that already exists results in an Internal Server Error

Comment (mattjackson220) [reply traffic_ops/traffic_ops_golang/routing/routes.go:423]: ill add in validation and changelog entries for it. this was necessary because the service category name is the primary key. Using the UpdateHandler doesnt work because it has to update using the previous information. So when i was using it, it returned that the service category doesnt exist because it was looking for the new name

Comment (rawlinp) [reply traffic_ops/traffic_ops_golang/routing/routes.go:423]: Ah, thanks for the info -- I didn't catch that on the first pass. That makes sense.

**Comment by mitchell852** [comment]
should there be any validation on name like alphanumerics and dashes?

**Review by mitchell852 — APPROVED** [review]
works as i'd expect. just the name validation question i had.

File: traffic_ops/traffic_ops_golang/servicecategory/servicecategories.go [hunk 2]
```diff
@@ -123,89 +118,78 @@ func (serviceCategory *TOServiceCategory) Create() (error, error, int) {
...
-			}
+	var origSC TOServiceCategory
+	if err := inf.Tx.QueryRow(`SELECT name FROM service_category WHERE name = $1`, name).Scan(&origSC.Name); err != nil {
+		if err == sql.ErrNoRows {
+			api.HandleErr(w, r, inf.Tx.Tx, http.StatusBadRequest, errors.New("no service category found with name "+name), nil)
+			return
 		}
-
-		return filteredSCByDS, nil, nil, errCode, maxTime
+		log.Errorf("MATT JACKSON in here")
```
Comment (rawlinp) [inline traffic_ops/traffic_ops_golang/servicecategory/servicecategories.go:157]: I think you forgot to remove this

File: traffic_ops/traffic_ops_golang/servicecategory/servicecategories.go [hunk 3]
```diff
@@ -123,89 +118,78 @@ func (serviceCategory *TOServiceCategory) Create() (error, error, int) {
...
+		api.HandleErr(w, r, inf.Tx.Tx, http.StatusInternalServerError, nil, err)
+		return
 	}
+	log.Errorf("MATT JACKSON or here")
```
Comment (rawlinp) [inline traffic_ops/traffic_ops_golang/servicecategory/servicecategories.go:161]: and this

File: traffic_ops/traffic_ops_golang/servicecategory/servicecategories.go [hunk 4]
```diff
@@ -123,89 +118,78 @@ func (serviceCategory *TOServiceCategory) Create() (error, error, int) {
...
-			return true
-		}
+	resp, err := inf.Tx.Tx.Exec(updateQuery(), newSC.Name, name)
+	if err != nil {
+		userErr, sysErr, errCode = api.ParseDBError(err)
+		api.HandleErr(w, r, inf.Tx.Tx, errCode, userErr, sysErr)
+		return
 	}
-	return false
+	api.CreateChangeLogRawTx(api.ApiChange, api.Updated+" Service Category: "+newSC.Name, inf.User, inf.Tx.Tx)
```
Comment (rawlinp) [inline traffic_ops/traffic_ops_golang/servicecategory/servicecategories.go:169]: Since we have the data, we could include the old name as well as the new name here.

**Review by rawlinp — CHANGES_REQUESTED** [review]

File: traffic_ops/traffic_ops_golang/servicecategory/servicecategories.go (hunk 2 above)
Comment (mattjackson220) [reply traffic_ops/traffic_ops_golang/servicecategory/servicecategories.go:157]: how embarrassing

File: traffic_ops/traffic_ops_golang/servicecategory/servicecategories.go (hunk 4 above)
Comment (rawlinp) [reply traffic_ops/traffic_ops_golang/servicecategory/servicecategories.go:166]: Whoops, I meant in the changelog entry

**Review by rawlinp — APPROVED** [review]
LGTM, works like a charm
